### PR TITLE
update wallet error handling

### PR DESCRIPTION
**Pull request by TimoGlastra**
Update the wallet error handling to be a bit less repetitive

﻿Signed-off-by: Timo Glastra <timo@animo.id>

### CODE REVIEW

File: src/lib/protocols/credentials/__tests__/StubWallet.ts [hunk 1]
```diff
@@ -4,6 +4,12 @@ import { UnpackedMessageContext } from '../../../types';
 
 export class StubWallet implements Wallet {
   private wh?: number | undefined;
+
+  public get walletHandle() {
```
Comment (jakubkoci) [inline src/lib/protocols/credentials/__tests__/StubWallet.ts:8]: Could we keep it private?

File: src/lib/protocols/credentials/__tests__/StubWallet.ts [hunk 2]
```diff
@@ -4,6 +4,12 @@ import { UnpackedMessageContext } from '../../../types';
 
 export class StubWallet implements Wallet {
   private wh?: number | undefined;
```
Comment (jakubkoci) [inline src/lib/protocols/credentials/__tests__/StubWallet.ts:6]: Maybe we can also rename it to `_walletHandle` to keep consistency with other getters. But I don't have a strong opinion here 🤷‍♂️

**Review by jakubkoci — COMMENTED** [review]
Nice refactor 👍  Just some details...

File: src/lib/protocols/credentials/__tests__/StubWallet.ts (hunk 1 above)
Comment (TimoGlastra) [reply src/lib/protocols/credentials/__tests__/StubWallet.ts:8]: I see the `SignatureDecoratorUtils.test.ts` uses it:

```ts
const verkey = await indy.createKey(wallet.walletHandle as number, { seed: seed1 });
```

The test compares the result from the wallet to a native indy call. So this woud need a refactor. It was already public. Or do you mean just the `StubWallet`?

Comment (TimoGlastra) [reply src/lib/protocols/credentials/__tests__/StubWallet.ts]: I left `walletHandle` in `IndyWalet` public for now. but applied the rest of your feedback

File: src/lib/protocols/credentials/__tests__/StubWallet.ts (hunk 2 above)
Comment (TimoGlastra) [reply src/lib/protocols/credentials/__tests__/StubWallet.ts]: Updated

**Review by jakubkoci — APPROVED** [review]

**Pull request merged**